src/index.ts

```diff
@@ -1,22 +1,59 @@
-import express from "express";
+import { createServer, IncomingMessage } from "node:http";
+
+import { verifyAndParseRequest, createAckEvent } from "@copilot-extensions/preview-sdk";
 import OpenAI from "openai";
-import { verifySignatureMiddleware } from "./validate-signature.js";
+
 import { describeModel } from "./functions/describe-model.js";
 import { executeModel } from "./functions/execute-model.js";
 import { listModels } from "./functions/list-models.js";
 import { RunnerResponse } from "./functions.js";
 import { recommendModel } from "./functions/recommend-model.js";
 import { ModelsAPI } from "./models-api.js";
-const app = express();
 
-app.post("/", verifySignatureMiddleware, express.json(), async (req, res) => {
+const server = createServer(async (request, response) => {
+  if (request.method === "GET") {
+    response.statusCode = 200;
+    response.end(`OK`);
+    return;
+  }
+
+  const body = await getBody(request);
+
+  let verifyAndParseRequestResult: Awaited<ReturnType<typeof verifyAndParseRequest>>;
+  let apiKey: string;
+  try {
+    const signature = request.headers["github-public-key-signature"] as string;
+    const keyID = request.headers["github-public-key-identifier"] as string;
+    apiKey = request.headers["x-github-token"] as string;
```

Comment:
We can move the API key outside:

```suggestion
  const apiKey = request.headers["x-github-token"] as string;
  try {
    const signature = request.headers["github-public-key-signature"] as string;
    const keyID = request.headers["github-public-key-identifier"] as string;
```